sklearn/utils/validation.py

```diff
@@ -1275,13 +1275,25 @@ def _check_sample_weight(sample_weight, X, dtype=None):
        is be allocated.  If `dtype` is not one of `float32`, `float64`,
        `None`, the output will be of dtype `float64`.
 
+    force_positive : {True, False or None}
+        If True, 'sample_weight' will raise error on negative values.
+        If False, 'sample_weight' being negative won't raise an error.
+        If None, assumes value of assume_positive_sample_weights,
+        which is initially set to True.
+
     Returns
     -------
     sample_weight : ndarray, shape (n_samples,)
        Validated sample weight. It is guaranteed to be "C" contiguous.
     """
     n_samples = _num_samples(X)
 
+    if force_positive is None:
+        force_positive = _get_config()['assume_positive_sample_weights']
+        if force_positive is False:
+            warnings.warn("assume_positive_sample_weights=False - "
+            "negative values in sample_weight won't raise an error.")
```

Comment:
I assume setting ```force_positive``` to ```False``` will raise a warning.
This makes sure the warning is raised if we set ```force_positive``` to ```False``` when calling the function.
```suggestion
    if not force_positive:
        warnings.warn("assume_positive_sample_weights=False - "
        "negative values in sample_weight won't raise an error.")
```

Reply:
Sorry, @haochunchang, but I'm not sure what your point is. Does it change anything overall?